engines/efh/savegames.cpp

```diff
@@ -0,0 +1,201 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/savefile.h"
+#include "graphics/scaler.h"
+#include "graphics/thumbnail.h"
+#include "common/system.h"
+#include "graphics/palette.h"
+
+#include "efh/efh.h"
+
+namespace Efh {
+
+Common::String EfhEngine::getSavegameFilename(int slot) {
+	return _targetName + Common::String::format("-%03d.SAV", slot);
+}
+
+bool EfhEngine::canLoadGameStateCurrently() {
+	return true;
+}
+
+bool EfhEngine::canSaveGameStateCurrently() {
+	return _saveAuthorized;
+}
+
+Common::Error EfhEngine::loadGameState(int slot) {
+	Common::InSaveFile *saveFile = g_system->getSavefileManager()->openForLoading(getSaveStateName(slot));
+	if (!saveFile)
+		return Common::kReadingFailed;
+
+	uint32 signature = saveFile->readUint32LE();
```

Comment:
Instead of yet-another copy/paste bespoke save for the metadata. I'd recommend switching ti ExtendedSaves. It is one-liner these days.